kinesis.go

```diff
@@ -53,61 +65,62 @@ type streamStatusTypes map[int]string
 type kinesis struct {
 	stream            string
 	shard             string
+	endPoint          string
 	shardIteratorType string
 	shardIterator     string
 	sequenceNumber    string
 	sequenceNumberMu  sync.Mutex
 
-	client gokinesis.KinesisClient
+	client kinesisiface.KinesisAPI
 
 	msgCount int64
 	errCount int64
 }
 
 func (k *kinesis) init(stream, shard, shardIteratorType, accessKey, secretKey, region string) (*kinesis, error) {
 
-	auth, err := authenticate(accessKey, secretKey)
+	sess, err := authenticate(accessKey, secretKey)
+	conf := aws.NewConfig().WithRegion(region)
+	if k.endPoint != "" {
+		conf = conf.WithEndpoint(k.endPoint)
+	}
+	client := awsKinesis.New(sess, conf)
+
 	k = &kinesis{
 		stream:            stream,
 		shard:             shard,
 		shardIteratorType: shardIteratorType,
-		client:            gokinesis.New(auth, region),
+		client:            client,
 	}
 	if err != nil {
 		return k, err
 	}
 
 	err = k.initShardIterator()
-	if err != nil {
-		return k, err
-	}
-
-	return k, nil
+	return k, err
 }
 
-func (k *kinesis) args() *gokinesis.RequestArgs {
-	args := gokinesis.NewArgs()
-	args.Add("StreamName", k.stream)
-	args.Add("ShardId", k.shard)
-	args.Add("ShardIterator", k.shardIterator)
-
+func (k *kinesis) initShardIterator() error {
+	// log.Println(k.sequenceNumber)
```

Comment:
instead of being commented, should this be logged at a lower log level (debug or lower, if the logger supports it)?